[tflite] enable passing int32 elementray arith to NNAPI

freedomtan opened the pull request:
NNAPI 1.3 (Android API level 30) [1] supports int32 add, sub, mul, and div. This patch enables passing +-*/ with int32 tensors to NNAPI. With that, we can do something like
```
./mul_test   --gtest_filter=IntegerMulOpTest.NoActivation --use_nnapi=1
```
on Android R devices, such as Pixel 4 running Android R beta.

[1] https://developer.android.com/ndk/reference/group/neural-networks

freedomtan committed:
enable passing int32 elementray arith to NNAPI

miaowang14 reviewed (dismissed):
Thanks for your work!

rthadur commented:
@freedomtan this changes are breaking this test https://github.com/tensorflow/tensorflow/blob/master/tensorflow/lite/kernels/add_test.cc#L208 , can you please check ?

miaowang14 commented on tensorflow/lite/delegates/nnapi/nnapi_delegate.cc:
```diff
@@ -1553,7 +1615,11 @@ bool NNAPIDelegateKernel::Validate(
     case kTfLiteBuiltinAdd: {
       ExpectMaxOpVersion(version, 2, &val_ctx);
-      ExpectIsFloatOrQuant8Operator(context, node, &val_ctx);
+      if (android_sdk_version >= kMinSdkVersionForNNAPI13) {
```
NNAPI in Android R requires the fused activation function to be NONE:
https://android.googlesource.com/platform/frameworks/ml/+/refs/heads/master/nn/runtime/include/NeuralNetworks.h#263

Similarly for MUL, SUB, DIV.

freedomtan replied:
changes added. Thanks.

freedomtan committed:
revert DIV part

revert div op part because there is semantic gap between tflite
and nnapi.

freedomtan committed:
add checking for act functions for INT32 cases

checked we can pass all add, sub, and mul test `--with_nnapi=1`

```
./add_test  --use_nnapi=1
./sub_test  --use_nnapi=1
./mul_test  --use_nnapi=1
```

freedomtan commented on tensorflow/lite/delegates/nnapi/nnapi_delegate.cc:
```diff
@@ -1968,9 +2040,16 @@ bool NNAPIDelegateKernel::Validate(
+      const TfLiteType input_type =
+          context->tensors[node->inputs->data[0]].type;
+
+      if (android_sdk_version >= kMinSdkVersionForNNAPI13) {
```
reverted DIV part, thanks.

freedomtan commented:
@miaowang14 and @rthadur: I reverted DIV part and added activation function check, and ran add, sub, mul kernel tests with `--with_nnapi=1`
